app/scripts/controllers/mmi-controller.js

```diff
@@ -542,8 +542,11 @@ export default class MMIController extends EventEmitter {
     const getAccountDetails = (address) =>
       this.custodyController.getAccountDetails(address);
     const extensionId = this.extension.runtime.id;
+
+    const { networkConfigurations } = this.networkController.state;
+
     const networks = [
-      ...this.preferencesController.getRpcMethodPreferences(),
+      ...networkConfigurations,
```

Comment:
It seems that when we transitioned from PreferencesController to NetworkController we might have missed this controller. 😕 

That said, `networkConfigurations` is an object, not an array, so I'm not sure this would work like you expect. It's difficult to see here because this file is JavaScript, but you can see that in the `core` repo by looking at the [type for `networkConfigurations`](https://github.com/MetaMask/core/blob/2b67b9863120d17386f5f17eba3f176784ccb713/packages/network-controller/src/NetworkController.ts#L105).

Therefore, what you may want here is:

``` suggestion
    const { networkConfigurations: networkConfigurationsById } = this.networkController.state;
    const networkConfigurations = Object.values(networkConfigurationsById);

    const networks = [
      ...networkConfigurations,
```

Reply:
ahhhh well spotted @mcmire, let me change that 👍🏼